### PR TITLE
[DS-281]: feat: getAATextColor function added

### PR DESCRIPTION
**Pull request by SVonKarajann**
## Description

[Ticket](https://orfium.atlassian.net/browse/DS-281)

### CODE REVIEW

**Comment by panvourtsis** [comment]
Can we have the pale and color util fixed as well with this solution? 
Screenshot 2021-12-14 at 3 22 29 PM

File: src/theme/palette.ts [hunk 1]
```diff
@@ -117,76 +119,22 @@ export type formFieldStyles = 'filled' | 'outlined' | 'elevated';
...
-    500: ['darkGrey', 'magenta', 'purple', 'darkBlue', 'blue', 'darkOrange'],
-    450: ['darkGrey', 'purple', 'darkBlue', 'blue', 'darkOrange'],
-    400: ['darkGrey', 'darkOrange'],
-    350: ['darkGrey', 'darkOrange'],
-  };
-  const pickedShade = colorsForWhiteText[shade];
-  const pickedColor =
-    (pickedShade && pickedShade?.find(item => item === color)) || color === 'neutralBlack';
+): string => {
+  const palette = enhancePaletteWithShades(lightPaletteConfig);
```
Comment (Christos-Zacharopoulos) [inline src/theme/palette.ts:128]: I am a little concerned about performance here. We are calling `enhancePaletteWithShades` that's a recursive function two times in a row.

Comment (Christos-Zacharopoulos) [reply src/theme/palette.ts:128]: Took a quick look at the usages and where we use `pickTextColorFromSwatches` we have access to the theme. 

I suggest we pass the theme as a prop to `pickTextColorFromSwatches` and avoid the overhead of running `enhancePaletteWithShades` over and over again.

**Review by Christos-Zacharopoulos — CHANGES_REQUESTED** [review]

**Comment by panvourtsis** [comment]
@Christos-Zacharopoulos can I suggest trying anything new on a different scope? I mean that we are trying to remove the mapping of the colors and replace it with AA. We can create a new ticket with a scope to make this function more performant.  What do you say?

File: src/theme/palette.ts (hunk 1 above)
Comment (panvourtsis) [reply src/theme/palette.ts:128]: @Christos-Zacharopoulos can I suggest trying anything new on a different scope? I mean that we are trying to remove the mapping of the colors and replace it with AA. We can create a new ticket with a scope to make this function more performant. What do you say?

Comment (Christos-Zacharopoulos) [reply src/theme/palette.ts:128]: Works for me. 👍

**Review by Christos-Zacharopoulos — APPROVED** [review]
LGTM! 🎉

**Pull request merged**